pkg/core/children.go

```diff
@@ -142,6 +142,10 @@ func getChildNamesByType(obj podController) (map[string]configMetadata, map[stri
 	return configMaps, secrets
 }
 
+func isOptional(b *bool) bool {
```

Comment:
I'd be tempted to rename this.
```suggestion
func isRequired(b *bool) bool {
```
Thinking about the wording here, one would expect `isOptional` should return `true` if `cm.Optional == true` and `false` if `cm.Optional == false`, but we are negating the value of cm.Optional in this assertion, so it doesn't read right with the expectation IMO. If we rename the method to `isRequired` the logic now makes more sense and `required: isRequired(cm.Optional)` also flows better.

WDYT?

Reply:
Yes, I agree. I'll change that.